src/game.cpp

```diff
@@ -9515,331 +9503,279 @@ int game::list_items(const int iLastState)
-                    add_msg(m_info, _("You can't travel there."));
-                }
+        } else if( action == "TRAVEL_TO" ) {
+            if( !u.sees( u.pos() + active_pos ) ) {
+                add_msg(_("You can't see that destination."));
             }
-
-            if ( uistate.list_item_sort == 1 ) {
-                ground_items = ground_items_radius;
-            } else if ( uistate.list_item_sort == 2 ) {
-                std::sort( ground_items.begin(), ground_items.end(), map_item_stack::map_item_stack_sort );
+            auto route = m.route( u.pos(), u.pos() + active_pos, u.get_pathfinding_settings(), u.get_path_avoid() );
+            if( route.size() > 1 ) {
+                route.pop_back();
+                u.set_destination( route );
+                break;
+            } else {
+                add_msg(m_info, _("You can't travel there."));
             }
+        }
+        if ( uistate.list_item_sort == 1 ) {
+            ground_items = ground_items_radius;
+        } else if ( uistate.list_item_sort == 2 ) {
+            std::sort( ground_items.begin(), ground_items.end(), map_item_stack::map_item_stack_sort );
+        }
 
-            if (refilter) {
-                refilter = false;
+        if (refilter) {
+            refilter = false;
+            filtered_items = filter_item_stacks(ground_items, sFilter);
+            highPEnd = list_filter_high_priority(filtered_items, list_item_upvote);
+            lowPStart = list_filter_low_priority(filtered_items, highPEnd, list_item_downvote);
+            iActive = 0;
+            page_num = 0;
+            iLastActive = tripoint_min;
+            iItemNum = filtered_items.size();
+        }
 
-                filtered_items = filter_item_stacks(ground_items, sFilter);
-                highPEnd = list_filter_high_priority(filtered_items, list_item_upvote);
-                lowPStart = list_filter_low_priority(filtered_items, highPEnd, list_item_downvote);
-                iActive = 0;
-                page_num = 0;
-                iLastActive = tripoint_min;
-                iItemNum = filtered_items.size();
+        if ( addcategory ) {
+            addcategory = false;
+            iCatSortNum = 0;
+            mSortCategory.clear();
+            if ( highPEnd > 0 ) {
+                mSortCategory[0] = _("HIGH PRIORITY");
+                iCatSortNum++;
             }
-
-            if ( addcategory ) {
-                addcategory = false;
-                std::string sLastCategoryName = "";
-                iCatSortNum = 0;
-                mSortCategory.clear();
-
-                if ( highPEnd > 0 ) {
-                    mSortCategory[0] = _("HIGH PRIORITY");
-                    iCatSortNum++;
+            std::string last_cat_name;
+            for( int i = std::max( 0, highPEnd ); i < std::min( lowPStart, ( int )filtered_items.size() ); i++ ) {
+                const std::string cat_name = filtered_items[i].example->get_category().name;
```

Comment:
Would be better if define `cat_name` as a const reference to avoid unnecessary copy?